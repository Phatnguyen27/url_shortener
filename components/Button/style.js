import styled from 'styled-components';

export const Wrapper = styled.button`
    height: ${(props) => props?.theme?.button?.height?.value};
    font-size: ${(props) => props?.theme?.button?.fontSize?.value};
    width: ${(props) => props?.theme?.button?.width?.value};
`;
