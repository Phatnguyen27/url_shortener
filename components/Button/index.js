import React from 'react';
import * as style from './style';

const Button = (props) => {
    const { onClickHandler, name, id } = props;
    const onclickHandler = () => {
        onClickHandler();
    }
    return <style.Wrapper id = {id} name = {name} onClick={onclickHandler}>{props.content}</style.Wrapper>
};

export default Button;