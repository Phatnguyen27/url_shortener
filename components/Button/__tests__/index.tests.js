import React from "react";
import renderer from "react-test-renderer";
import Button from "../";
import ReactDOM from "react-dom";
import TestUtils from "react-dom/test-utils";

const testProps = {
    id: "id",
    name: "name",
    content: "button",
    onClickHandler: {
        onClickHandler: jest.fn(() => ({}))
    }
}
const component = renderer.create(
    <Button {...testProps}/>
);
const tree = component.toJSON();

describe("test button component", () => {
    it ("should render correctly", () => {
        expect(tree).toMatchSnapshot();
    })
    it ("should recognise the click event", () => {
        let container;
        const click = jest.fn();
        container = document.createElement("div");
        document.body.appendChild(container);
        ReactDOM.render(<Button {...testProps} onClickHandler={click}/>, container);
        const button = container.querySelector("button");
        TestUtils.act(() => {
            button.dispatchEvent(new MouseEvent('click', {bubbles: true}));
          });
        expect(click).toHaveBeenCalled();
    })
})