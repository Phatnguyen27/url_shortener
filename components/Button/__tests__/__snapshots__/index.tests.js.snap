// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> 1`] = `
<button
  className="sc-bdfBwQ leIcBc"
  id="id"
  name="name"
  onClick={[Function]}
>
  button
</button>
`;

exports[`test button component should render correctly 1`] = `
<button
  className="sc-bdfBwQ leIcBc"
  id="id"
  name="name"
  onClick={[Function]}
>
  button
</button>
`;
